packages/lib/src/classes/ApplicationCommand.ts

```diff
@@ -0,0 +1,33 @@
+import {
+	ApplicationCommandOptionData,
+	ApplicationCommandType,
+	AutocompleteFocusedOption,
+	AutocompleteInteraction,
+	ChatInputCommandInteraction,
+	MessageContextMenuCommandInteraction,
+	UserContextMenuCommandInteraction
+} from "discord.js"
+import { BetterClient, ApplicationCommandOptions } from "../index.js"
+import BaseComponent from "./_BaseComponent.js"
+
+export default class ApplicationCommand extends BaseComponent {
+	public readonly description: string
+	public readonly type: ApplicationCommandType = ApplicationCommandType.ChatInput
+	public readonly options: ApplicationCommandOptionData[] = []
+
+	constructor(key: string, client: BetterClient, options: ApplicationCommandOptions) {
+		super(key, client, options)
+		this.description = options.description
+		this.options = options.options || []
+		this.type = options.type || ApplicationCommandType.ChatInput
+	}
+
+	public override async run(
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
not exclusive to just this line but if we could figure out the shit ton of eslint ignores particularly in the classes the code would look tidier.